File: gateway-server/src/main/java/org/lucycato/gatewayserver/application/port/in/command/LogResponseCommand.java
package org.lucycato.gatewayserver.application.port.in.command;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.lucycato.common.SelfValidating;

import java.util.List;
import java.util.Map;

@Getter
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class LogResponseCommand extends SelfValidating<LogResponseCommand> {
    @NotNull
    private Integer statusCode;

    @NotEmpty
    private Map<String, List<String>> headers;

    public LogResponseCommand(Integer statusCode, Map<String, List<String>> headers) {
        this.statusCode = statusCode;
        this.headers = headers;

        this.validateSelf();
    }
}

File: gateway-server/src/main/kotlin/org/lucycato/gatewayservice/application/port/in/commnad/LogResponseCommand.kt
package org.lucycato.gatewayservice.application.port.`in`.commnad

import jakarta.validation.constraints.NotEmpty
import org.lucycato.common.SelfValidating

data class LogResponseCommand(
        val statusCode: Int,

        @field:NotEmpty
        val headers: Map<String, List<String>>
): SelfValidating<LogResponseCommand>() {
    init {
        validateSelf()
    }
}